lib/modules/manager/gomod/artifacts.ts

```diff
@@ -134,9 +134,9 @@ export async function updateArtifacts({
   const vendorDir = upath.join(goModDir, 'vendor/');
   const vendorModulesFileName = upath.join(vendorDir, 'modules.txt');
   const useVendor =
-    !config.postUpdateOptions?.includes('gomodSkipVendor') &&
-    (await readLocalFile(vendorModulesFileName)) !== null;
-
+    !!config.postUpdateOptions?.includes('gomodVendor') ||
+    ((await readLocalFile(vendorModulesFileName)) !== null &&
+      !config.postUpdateOptions?.includes('gomodSkipVendor'));
```

Comment:
```suggestion
    !!config.postUpdateOptions?.includes('gomodVendor') ||
    (!config.postUpdateOptions?.includes('gomodSkipVendor') && 
      (await readLocalFile(vendorModulesFileName)) !== null);
```

Restore original ordering

Reply:
sure thing, will fix the tests too